api/src/pcapi/scripts/delete_offer_description/main.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import datetime
+import logging
+import statistics
+import time
+
+import pytz
+from sqlalchemy import func
+from sqlalchemy import text
+
+import pcapi.core.offers.models as offers_models
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+BATCH_SIZE = 1_000
+REPORT_EVERY = 1_000
+
+
+def _get_eta(end: int, current: int, elapsed_per_batch: list[int]) -> str:
+    left_to_do = end - current
+    seconds_eta = left_to_do / BATCH_SIZE * statistics.mean(elapsed_per_batch)
+    eta = datetime.datetime.utcnow() + datetime.timedelta(seconds=seconds_eta)
+    eta = eta.astimezone(pytz.timezone("Europe/Paris"))
+    str_eta = eta.strftime("%d/%m/%Y %H:%M:%S")
+    return str_eta
+
+
+def delete_description(starting_id: int, ending_id: int, not_dry: bool = False) -> None:
+    max_id = db.session.query(func.max(offers_models.Offer.id)).scalar()
+    vacuum_treshold = int(max_id * 0.19)
+    logger.info("Run a vacuum every %s offers", vacuum_treshold)
+    elapsed_per_batch = []
+    to_report = 0
+    for i in range(starting_id, ending_id, BATCH_SIZE):
+        start_time = time.perf_counter()
+        db.session.execute(
+            """
+            update offer set "description" = null
+            where id between :start and :end and "productId" is not null
+            """,
+            params={"start": i, "end": i + BATCH_SIZE},
+        )
+        if not_dry:
+            db.session.commit()
+            if i % vacuum_treshold:
```

Comment:
Ça ne serait pas
```suggestion
            if i % vacuum_treshold == 0:
```

plutôt ? Sinon on va vacuum à chaque offre ?

Reply:
Oui, en effet ça faisait un vacuum par batch, toutes les 1000 offres en local 😬